packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,21 +1,14 @@
-//! Provides a memory format similar to Arrows `Field`, `DataType`, and `Schema`.
+//! Provides datastructures to organize field specifications and data types.
 //!
-//! As only a selection of Arrow [`DataType`]s is supported at hEngine and additional constraints on
-//! the Arrow column names are required to distinguish the different [`FieldSource`]s and
-//! [`FieldScope`]s, this module abstracts over Arrow types. It defines [`FieldType`], which uses
-//! Arrow types underneath and adds an `Any` type, which can represent arbitrary data and
-//! [`FieldSpec`] to associate a name with the [`FieldType`].
+//! Each supported data type is represented by a [`FieldType`]. It's associated with a name in
+//! [`FieldSpec`]. Depending on the underlying [`FieldTypeVariant`], types may be nested.
```

Comment:
Perhaps something like, Fields are defined by their [`FieldSpec`], which encapsulates their name and [`FieldType`]. Fields can be nested depending on their [`FieldTypeVariant`]